app/components/super-features/super-features-relations/super-features-relations-modal/super-features-relations-modal.tests.js

```diff
@@ -0,0 +1,204 @@
+'use strict';
+
+describe('Directive: superFeaturesRelationsModalOpener', function () {
+
+  var $;
+  var $modal;
+  var $q;
+  var $rootScope;
```

Comment:
Instead of using `$rootScope` in this test I would suggest using a variable called `$parentScope` and setting that to `$rootScope.$new()` in `inject`. Then, in the `directiveBuilderWithDynamicHtml` call use `$parentScope` and make modifications to `$parentScope` instead of `$rootScope` in your tests.

This will effectively isolate scope changes to these tests instead of potentially polluting the `$rootScope`.

Note then you can call `$parentScope.$digest()` to digest only the `$parentScope`. As opposed to digesting the entire angular app with `$rootScope.$digest()`.